src/routes/v1/secret.go

```diff
@@ -43,57 +43,75 @@ func secretHandler(handler func(controller controllers.SecretController, c *gin.
 
 // CreateSecret creates a new secret in a specific namespace.
 func CreateSecret() gin.HandlerFunc {
-	return secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
-		namespaceName := c.Param("namespaceName")
-
+	return func(c *gin.Context) {
 		var request types.CreateSecretRequest
-		if err := c.ShouldBindJSON(&request); err != nil {
-			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-			return nil, err
+		if err := c.BindJSON(&request); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Invalid request", "details": err.Error()})
+			return
 		}
 
-		return controller.CreateSecret(namespaceName, request)
-	})
+		secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
+			return controller.CreateSecret(request)
+		})(c)
+	}
 }
 
 // GetSecrets gets all secrets in a specific namespace.
 func GetSecrets() gin.HandlerFunc {
-	return secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
-		namespaceName := c.Param("namespaceName")
-		return controller.GetSecrets(namespaceName)
-	})
+	return func(c *gin.Context) {
+		var request types.GetSecretsRequest
+		if err := c.BindUri(&request); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Invalid request", "details": err.Error()})
+			return
+		}
+
+		secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
+			return controller.GetSecrets(request)
+		})(c)
+	}
 }
 
 // GetSecret gets a specific secret from a specific namespace.
 func GetSecret() gin.HandlerFunc {
-	return secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
-		namespaceName := c.Param("namespaceName")
-		secretName := c.Param("secretName")
-		return controller.GetSecret(namespaceName, secretName)
-	})
+	return func(c *gin.Context) {
+		var request types.GetSecretRequest
+		if err := c.BindUri(&request); err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Invalid request", "details": err.Error()})
+			return
+		}
+
+		secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
+			return controller.GetSecret(request)
+		})(c)
+	}
 }
 
 // PatchSecret patches a specific secret in a specific namespace.
 func PatchSecret() gin.HandlerFunc {
-	return secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
-		namespaceName := c.Param("namespaceName")
-		secretName := c.Param("secretName")
-
+	return func(c *gin.Context) {
 		var request types.PatchSecretRequest
-		if err := c.ShouldBindJSON(&request); err != nil {
-			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-			return nil, err
+		if err := c.BindJSON(&request); err != nil {
```

Comment:
there`s no validation to URI here.  and the route should accept namespace name and secret name. you should divide to two struct and unify them after two validations function